skimage/restoration/tests/test_restoration.py

```diff
@@ -11,46 +12,60 @@
 test_img = util.img_as_float(camera())
 
 
-def test_wiener():
-    psf = np.ones((5, 5)) / 25
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_wiener(dtype):
+    psf = np.ones((5, 5), dtype=dtype) / 25
     data = convolve2d(test_img, psf, 'same')
     np.random.seed(0)
     data += 0.1 * data.std() * np.random.standard_normal(data.shape)
+    data = data.astype(dtype, copy=False)
     deconvolved = restoration.wiener(data, psf, 0.05)
+    assert deconvolved.dtype == dtype
```

Comment:
similar thought as above: won't this fail if `dtype` and `_float_type(dtype)` don't match?